Modules/Module_Processing/xASL_init_DefaultEffectiveResolution.m

```diff
@@ -48,6 +48,17 @@
     return;
 end
 
+% Obtain the x.Q.MRAcquisitionType field from the json
+if ~isempty(tJson) && isfield(tJson, 'Q') && isfield(tJson.Q, 'MRAcquisitionType')
+	x.Q.MRAcquisitionType = tJson.Q.MRAcquisitionType;
+end
+
+% If sequence is still missing we skip this function
```

Comment:
Will it not be defaulted to something that is potentially wrong? Not sure if we should always avoid crashing in case of "illegal" data.

Reply:
Previously, we were defaulting to 3DGrase or something. And you decided to remove this defaulting. 

I'm happy to insert "crash" here if you want...

Reply:
I would say that we need a single check that either MRAcquisitionType or PulseSequenceType is missing, and then we issue an error. Because this should already be covered by BIDS.

We can also add this to our generic parameter check at the beginning of the ASL module, then you can remove the check here (because submodule you start the ASL module with, it always needs to pass this generic parameter check).

Reply:
We are already checking the parameter presence at the start of ASL modules.

I've changed the code to an error. So in case they are missing, I'm issuing a full error. But that's a safety catch cause that situation wouldn't happen here.